libs/langchain/langchain/document_loaders/sharepoint.py

```diff
@@ -0,0 +1,59 @@
+"""Loader that loads data from Sharepoint Document Library"""
+from __future__ import annotations
+
+from typing import Iterator, List, Optional, Sequence
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base_o365 import (
+    O365BaseLoader,
+    _FileType,
+)
+from langchain.document_loaders.parsers.registry import get_parser
+from langchain.pydantic_v1 import Field
+
+
+class SharePointLoader(O365BaseLoader):
+    """Load  from `SharePoint`."""
+
+    document_library_id: str = Field(...)
+    """ The ID of the SharePoint document library to load data from."""
+    folder_path: Optional[str] = None
+    """ The path to the folder to load data from."""
+    object_ids: Optional[List[str]] = None
+    """ The IDs of the objects to load data from."""
+
+    @property
+    def _file_types(self) -> Sequence[_FileType]:
```

Comment:
The new blob loaders abstraction helps to prevent hard-coding knowledge of parsing into content fetching. This makes the loading code a lot easier to reuse.

The loader should take 2 attributes that should be part of the initializer. 

1. Blob parser
2. extensions